src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.service.ts

```diff
@@ -1,66 +1,62 @@
 import { Injectable } from '@angular/core';
 import { map } from 'rxjs/operators';
-import { Observable, lastValueFrom } from 'rxjs';
+import { BehaviorSubject, Observable } from 'rxjs';
 import { HttpClient, HttpParams, HttpResponse } from '@angular/common/http';
 import { Ide, IdeMappingDTO } from 'app/shared/user-settings/ide-preferences/ide.model';
 import { ProgrammingLanguage } from 'app/entities/programming/programming-exercise.model';
 
 @Injectable({ providedIn: 'root' })
 export class IdeSettingsService {
-    public readonly ideSettingsUrl = 'api/ide-settings';
+    public ideSettingsUrl = 'api/ide-settings';
+    public static readonly fallbackIde: Ide = { name: 'VS Code', deepLink: 'vscode://vscode.git/clone?url={cloneUrl}' };
+    private predefinedIdes: Ide[] = [IdeSettingsService.fallbackIde];
+    private idePreferencesSubject: BehaviorSubject<Map<ProgrammingLanguage, Ide>> = new BehaviorSubject<Map<ProgrammingLanguage, Ide>>(
+        new Map([[ProgrammingLanguage.EMPTY, IdeSettingsService.fallbackIde]]),
+    );
+    error?: string;
 
-    // cached value of the ide preferences to avoid unnecessary requests to the server
-    private idePreferences?: Map<ProgrammingLanguage, Ide>;
+    constructor(private http: HttpClient) {
+        // initially load predefined ides and ide preferences
+        this.loadPredefinedIdes().subscribe();
+        this.loadIdePreferences().subscribe();
+    }
 
-    constructor(private http: HttpClient) {}
+    public getPredefinedIdes(): Ide[] {
+        return this.predefinedIdes;
+    }
 
     /**
      * GET call to the server to receive the stored ide preferences of the current user
      * @return the saved ide preference which were found in the database or error
      */
     public loadPredefinedIdes(): Observable<Ide[]> {
-        return this.http.get<Ide[]>(this.ideSettingsUrl + '/predefined');
+        return this.http.get<Ide[]>(this.ideSettingsUrl + '/predefined').pipe(
+            map((ides) => {
+                this.predefinedIdes = ides;
+                return ides;
+            }),
```

Comment:
I think you should use tap instead of map here, since you return the exact input anyway. This way you can get rid of the return